core/src/components/ripple-effect/test/basic/ripple-effect.e2e.ts

```diff
@@ -0,0 +1,63 @@
+import { expect } from '@playwright/test';
+import type { IonicPage } from '@utils/test/playwright';
+import { test } from '@utils/test/playwright';
+
+test.describe('ripple-effect: basic', () => {
+  test('should add .ion-activated when pressed', async ({ page }) => {
+    await verifyRippleEffect(page, '#small-btn');
+    await verifyRippleEffect(page, '#large-btn');
+    await verifyRippleEffect(page, '#large-btn-outline');
+    await verifyRippleEffect(page, '#large-btn-clear');
+    await verifyRippleEffect(page, '.block');
+  });
+
+  test.describe('ripple effect with nested ion-button', () => {
+    test('should add .ion-activated when the block is pressed', async ({ page }) => {
+      await page.goto(`/src/components/ripple-effect/test/basic?ionic:_testing=false&ionic:mode=md`);
+
+      const el = page.locator('#ripple-with-button');
+
+      await el.scrollIntoViewIfNeeded();
+
+      const boundingBox = await el.boundingBox();
+
+      if (boundingBox) {
+        await page.mouse.move(boundingBox.x + 5, boundingBox.y + 5);
+        await page.mouse.down();
+      }
+
+      // Waits for the ripple effect to be added
+      await page.waitForSelector('.ion-activated');
+
+      const elHandle = await el.elementHandle();
+      const classes = await elHandle?.evaluate((el) => el.classList.value);
+      expect(classes).toMatch('ion-activated');
+    });
+
+    test('should add .ion-activated when the button is pressed', async ({ page }) => {
+      await verifyRippleEffect(page, '#ripple-with-button ion-button');
+    });
+  });
+});
+
+const verifyRippleEffect = async (page: IonicPage, selector: string) => {
+  await page.goto(`/src/components/ripple-effect/test/basic?ionic:_testing=false&ionic:mode=md`);
+
+  const el = page.locator(selector);
+
+  await el.scrollIntoViewIfNeeded();
+
+  const boundingBox = await el.boundingBox();
+
+  if (boundingBox) {
+    await page.mouse.move(boundingBox.x + boundingBox.width / 2, boundingBox.y + boundingBox.height / 2);
+    await page.mouse.down();
+  }
+
+  // Waits for the ripple effect to be added
+  await page.waitForSelector(`${selector}.ion-activated`);
```

Comment:
If I check the button demo on https://ionicframework.com/docs, the button receives `ion-activated` for a brief moment on click, even with the busted ripple effect. Does the test fail with the broken state? 🤔

Reply:
Are you possibly being served from a cache? I don't see the class added to the button element when clicking and holding on the button in the docs examples. Chrome also doesn't indicate that the `class` attribute was modified during mouse down.

Reply:
Strange that doesn't happen on Mac 🤔  https://user-images.githubusercontent.com/13732623/163049314-6e795818-897e-464b-ba0c-ba589f0bd4fe.mp4

Anyways, I confirmed that with the changes on `main`, the test suite fails. With the changes in this branch, the test suite passes.

Reply:
Cool, that's what's important 👍